### PR TITLE
Make test_base_opencl framework respect SYCL device selector choice

### LINKED ISSUE #21: Update CTS for stream interface clarifications
Tracking of CTS investigation for spec MR: https://github.com/KhronosGroup/SYCL-Docs/pull/58

### PR DESCRIPTION
**Pull request by bader**
Fixes #21

### CODE REVIEW

File: util/test_base_opencl.cpp [hunk 1]
```diff
@@ -33,64 +35,32 @@ bool test_base_opencl::setup(logger &log) {
...
-    FAIL(log, "Unable to retrieve list of platforms via clGetPlatformIDs()");
-    return false;
-  }
+  cts_selector ctsSelector;
+  const auto ctsContext = util::get_cts_object::context(ctsSelector);
 
-  platforms.reset(new cl_platform_id[N]);
-  if (platforms.get() == nullptr) {
-    FAIL(log, "Malloc failed to allocate memory for platform list");
+   if ( ctsContext.get_devices().size() < 1) {
```
Comment (keryell) [inline util/test_base_opencl.cpp]: Weird spacing

Comment (bader) [reply util/test_base_opencl.cpp]: Thanks!
Fixed.

**Comment by ProGTX** [comment]
I've looked into this and the selector chooses a host device when running the test on the host, and then the `.get()` function throws. This affects the `buffer`, `kernel`, `opencl_interop`, and `program` tests. I was hoping that updating the branch to https://github.com/KhronosGroup/SYCL-CTS/pull/24/ would fix this, but it doesn't, `test_base_opencl::setup` is still being called for the host device.

**Comment by bader** [comment]
> I've looked into this and the selector chooses a host device when running the test on the host, and then the `.get()` function throws.

I think we can skip setup if the selector chooses a host device. Right?

> This affects the `buffer`, `kernel`, `opencl_interop`, and `program` tests. I was hoping that updating the branch to #24 would fix this, but it doesn't, `test_base_opencl::setup` is still being called for the host device.

Hm... I expect only `opencl_inteop` test to be affected as #24 replaced `test_base_opencl::setup` to `test_base::setup` for `buffer`, `kernel` and `program` tests. Did I miss something?

**Comment by bader** [comment]
@ProGTX, I've updated the patch. Please, let me know if you have any issues with it.

**Comment by bader** [comment]
@ProGTX, sorry for the mess, but I somehow missed the fix. Please, check https://github.com/KhronosGroup/SYCL-CTS/commit/40fe5d21f5a578a196d2822de7cdbb41612edd38.

**Comment by bader** [comment]
@AerialMantis, great news. Thanks for the update.
@keryell, could you approve updated version of the patch, please?

File: util/test_base_opencl.cpp [hunk 2]
```diff
@@ -39,64 +41,35 @@ bool test_base_opencl::setup(logger &log) {
...
-
-  error =
-      clGetDeviceIDs(platforms.get()[0], CL_DEVICE_TYPE_ALL, 0, nullptr, &N);
-  if (!CHECK_CL_SUCCESS(log, error)) return false;
+  cts_selector ctsSelector;
+  const auto ctsContext = util::get_cts_object::context(ctsSelector);
 
-  if (N < 1) {
-    FAIL(log, "Unable to retrieve list of devices via clGetDeviceIDs()");
+  if (ctsContext.get_devices().size() < 1) {
```
Comment (keryell) [inline util/test_base_opencl.cpp]:
```suggestion
  if (ctsContext.get_devices().empty()) {
```

Comment (bader) [reply util/test_base_opencl.cpp]: Done.

**Pull request merged**